src/librbd/exclusive_lock/ReleaseRequest.cc

```diff
@@ -130,22 +131,51 @@ Context *ReleaseRequest<I>::handle_flush_notifies(int *ret_val) {
   ldout(cct, 10) << __func__ << dendl;
 
   assert(*ret_val == 0);
-  send_close_journal();
+
+  send_append_journal_event();
   return nullptr;
 }
 
 template <typename I>
-void ReleaseRequest<I>::send_close_journal() {
-  {
-    RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
-    std::swap(m_journal, m_image_ctx.journal);
-  }
+void ReleaseRequest<I>::send_append_journal_event(){
+    CephContext *cct = m_image_ctx.cct;
+    ldout(cct, 10) << __func__ << dendl;
+    {
+      RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
+      std::swap(m_journal, m_image_ctx.journal);
+    }
 
-  if (m_journal == nullptr) {
-    send_close_object_map();
-    return;
-  }
+    if (m_journal == nullptr) {
+       send_close_object_map();
+       return;
+    }
+
+    using klass = ReleaseRequest<I>;
+    Context *ctx = create_context_callback<
+      klass, &klass::handle_append_journal_event>(this);
+    //append journal event
+    {
+      RWLock::WLocker snap_locker(m_image_ctx.snap_lock);
+      m_image_ctx.journal->append_lock_release_event(ctx);
```

Comment:
Using m_journal instead of m_image_ctx.journal, I get error at L#87 in test/librbd/exclusive_lock/test_mock_ReleaseRequest.cc. It says expected to be called once but never called.
Am i missing anything? Do I need to invoke anywhere else?